Test whether github action runner can compile my C++ code

gxuu opened the pull request:
as title suggests

gxuu commented:
This PR does the following things:
- It packs the current server into a `.so`.
- It makes linter.yml runs it. So that we can see whether (a) CMake can run on Github Actions and (b) Whether we have an up to date C++ compiler.

Do not merge this PR, as building the C++ code should not be part of the linter workflow. Also, it's not a good idea to just copy and paste most of the code from `server.cpp` straight to `server_as_so.cpp`.

gxu

rafa-be commented on .github/workflows/linter.yml:
```diff
@@ -28,6 +28,15 @@ jobs:
+          sudo apt-get upgrade -y
+          sudo apt-get install libboost-all-dev -y
+          sudo apt-get install capnproto -y
+          pip install cmake
```
Should we install cmake through PIP instead of aptitude?

gxuu replied:
Yes probably. This PR is for testing

rafa-be replied:
I know, I'm trying to build your latest ObjectStorage PR myself, and this helps :)

rafa-be commented on scaler/CMakeLists.txt:
```diff
@@ -1,5 +1,16 @@
+# find_package(CapnProto CONFIG REQUIRED)
+# message(STATUS "Found CapnProto")
 
 add_subdirectory(object_storage)
```
I think this should be `add_subdirectory(scaler/object_storage)`

rafa-be replied:
Nevermind, I thought this was `/CMakeLists.txt`, not `/scaler/CMakeLists.txt"

gxuu commented on .flake8:
```diff
@@ -0,0 +1,7 @@
+    migrations
+    __pycache__
+    ./build
+    ./capnproto-c++-1.1.0
```
done

gxuu commented on download_dependencies.sh:
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
done

gxuu commented:
- added flake8 wildcard exclude list
- added version parametrization
- Made a script directory and put script there
- cleaned up cmake

Ready to merge

sharpener6 commented:
@gxuu please change `server_as_so.[h|cpp]` back, there is no reason to name as that

sharpener6 commented on .flake8:
```diff
@@ -0,0 +1,10 @@
+[flake8]
+exclude =
```
merge those flake8 options into `pyproject.toml` and remove this `.flake8` file

sharpener6 replied:
`$ pip install pflake8` and use `$ pflake8 .` to test

gxuu replied:
`pip install` sadly does not work on my system. 
But it should be fine now.
